recipe/build.sh

```diff
@@ -1,92 +1,112 @@
-    -DPYTHON_DEFAULT_EXECUTABLE="${SYS_PYTHON}"                             \
-    -DPYTHON_NATIVE_INTERPRETERS="${PYTHON_INTERPS}"                        \
-    -DINSTALL_PYTHON_EXAMPLES=1                                             \
-    -DENABLE_PYLINT=1                                                       \
-    -DENABLE_FLAKE8=1                                                       \
-    -DOPENCV_EXTRA_MODULES_PATH="../opencv_contrib-${PKG_VERSION}/modules"  \
-    "${CMAKE_EXTRA_ARGS[@]}"
+# append dependencies to CMAKE_EXTRA_ARGS
+for dep in EIGEN FFMPEG GSTREAMER OPENBLAS PROTOBUF QT;do
+    varname=WITH_${dep}
+    CMAKE_EXTRA_ARGS+=("-D${varname}=${!varname}")
+done
+# append debug args
+CMAKE_EXTRA_ARGS+=("${CMAKE_DEBUG_ARGS[@]}")
+echo "CMake_EXTRA_ARGS : ${CMAKE_EXTRA_ARGS[@]}"
 
-  if [[ ! $? ]]; then
-    echo "configure failed with $?"
-    exit 1
-  fi
+#TODO: check that libpng is found or use conda forge hack
 
+#export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$PREFIX/lib/pkgconfig
+#export PKG_CONFIG_LIBDIR=$PREFIX/lib
+mkdir -p build
+cd build
+cmake .. -LAH -GNinja                                                     \
+  ${CMAKE_ARGS}                                                           \
+  "${CMAKE_EXTRA_ARGS[@]}" `#includes platform specific deps and options` \
+  "${PYTHON_CMAKE_ARGS[@]}"                                               \
+  -DBUILD_DOCS=0                                                          \
+  -DBUILD_JASPER=0                                                        \
+  -DBUILD_JPEG=0                                                          \
+  -DBUILD_LIBPROTOBUF_FROM_SOURCES=0                                      \
+  -DBUILD_OPENEXR=ON                                                      \
+  -DBUILD_PERF_TESTS=0                                                    \
+  -DBUILD_PNG=0                                                           \
+  -DBUILD_PROTOBUF=0                                                      \
+  -DBUILD_TESTS=0                                                         \
+  -DBUILD_TIFF=0                                                          \
+  -DBUILD_ZLIB=0                                                          \
+  -DBUILD_opencv_apps=OFF `# issue linking with opencv_model_diagnostics` \
+  -DCMAKE_BUILD_TYPE="Release"                                            \
+  -DCMAKE_CROSSCOMPILING=ON         `# may not need`                      \
+  -DENABLE_CONFIG_VERIFICATION=ON                                         \
+  -DENABLE_FLAKE8=0                                                       \
+  -DENABLE_PYLINT=0      `# used for docs and examples`                   \
+  -DINSTALL_C_EXAMPLES=OFF                                                \
+  -DINSTALL_PYTHON_EXAMPLES=ON                                            \
+  -DOPENCV_DOWNLOAD_PATH="${SYS_PREFIX}"/conda-bld/src_cache              \
```

Comment:
This line needs to put things somewhere else than `${SYS_PREFIX}/conda-bld`. On non-docker builds on prefect that directory is shared between all builds and having anything in there causes conda build to fail.

Reply:
fixed. thanks!